src/rpc/util.cpp

```diff
@@ -793,9 +793,15 @@ UniValue RPCHelpMan::GetArgMap() const
     for (int i{0}; i < int(m_args.size()); ++i) {
         const auto& arg = m_args.at(i);
         std::vector<std::string> arg_names = SplitString(arg.m_names, '|');
+        RPCArg::Type argtype = arg.m_type;
```

Comment:
But if there's only one name here, it's unclear what name the bool type is for? Or we're assuming it can only be used as a positional param at that point?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/24963#discussion_r1467815704

> But if there's only one name here, it's unclear what name the bool type is for? Or we're assuming it can only be used as a positional param at that point?

Yes, that's a good point. The change I suggested to `wallet/rpc/spend.cpp` makes the code less clear, even if it is removes some code that is technically unnecessary. So your version is better.

In case anybody is following this though, the github comment this is responding to was mostly about `RPCHelpMan::GetArgMap`, which can still be simplified.